internal/pkg/template/workload.go

```diff
@@ -252,6 +255,48 @@ func (h HTTPHealthCheckOpts) IsHTTPS() bool {
 	return h.Port == "443"
 }
 
+type importable interface {
+	RequiresImport() bool
+}
+
+// Variable represents the value of an environment variable.
+type Variable interface {
+	importable
+	Value() string
+}
+
+// ImportedVariable returns a Variable that should be imported from a stack.
+func ImportedVariable(name string) Variable {
+	return envVar{
+		importName: name,
+	}
+}
+
+// PlainVariable returns a Variable that is a plain string value.
+func PlainVariable(value string) Variable {
+	return envVar{
+		value: value,
+	}
+}
+
+type envVar struct {
+	value      string
+	importName string
+}
+
+// RequiresImport returns true if the environment variable is imported from a stack.
+func (v envVar) RequiresImport() bool {
+	return v.importName != ""
```

Comment:
nit: I wonder if instead of putting two mutually exclusive fields in a single struct we can split the struct:
```go
type plainEnvVar string
func (var plainEnvVar) RequiresImport() bool { return false }
func (var plainEnvVar) Value() string { return var }

type importedEnvVar string
func (var importedEnvVar) RequiresImport() bool { return true }
func (var importedEnvVar) Value() string { return var } 
```